### ISSUE TITLE
Anyway to define an experiment?

### ISSUE THREAD

**Issue by hsingchien** [DynaSim/DynaSim#583]
Anyway to define an experiment and run it? e.g. inject certain amount of current to certain cell at a defined point (instead of a constant Iapp in the population equations)?

**Comment by kupiqu** [comment]
Why don't you create a vector instead of scalar value with whatever values you'd like in there? 

For instance to target a single specific unit you can use a vector of all zeros with the only exception of the cell position you want, and add that position whatever current you want to inject.

**Comment by kupiqu** [comment]
and if you want to make it vary in time, instead of a constant value, you will need to set a function of t (time)

**Comment by hsingchien** [comment]
> and if you want to make it vary in time, instead of a constant value, you will need to set a function of t (time)

Thanks! I'll try the vector. 
Would it be a more straightforward solution to have an explicit module to control these functions? Sort of like the 'electrode' window of NEURON, if I remember correctly.

**Comment by kupiqu** [comment]
Yes, I totally agree. it would be great to have easy ways to "design" experiments in DynaSim.

**Comment by benpolletta** [comment]
For a low-level electrophysiological experiment, the 'mechanism' framework
provides a pretty good solution - it is easy to create a mechanism that
constructs and applies any special input current to the cell, for example
by modifying the mechanism for tonic input.

**Comment by kupiqu** [comment]
This is (I think) what I described above wrt the original question. I think @hsingchien would like something more user friendly. He mentioned about 'electrode' in NEURON. Is anybody familiar with it?

**Comment by hsingchien** [comment]
![image](https://user-images.githubusercontent.com/15225459/66595925-907d5a00-eb61-11e9-877d-a7f4ed653145.png)
NEURON has a built-in clamp module as shown in this screen shot. More complex electrodes can be defined in the code/command line. I think what @benpolletta suggest is by far the best solution. A custom mechanism would do the job. Then however, another problem is how to apply the mechanism to only a subset of neurons in a population? Obviously the target cell can be separated and be its own population, but that is not very elegant. This might be solved by feeding an input mask, it seems. I am trying to figure out how that works.

**Comment by kupiqu** [comment]
Well, I guess I failed to see that you didn't have a specific mechanism implemented for the injected current, which certainly is what you should do, as @benpolletta suggested.

But once you have the mechanism, all what I explained above about the vector array should apply.
